cljfmt/src/cljfmt/core.cljc

```diff
@@ -41,16 +41,16 @@
   (nil? (z/up* zloc)))
 
 (defn- top? [zloc]
-  (and zloc (not= (z/node zloc) (z/root zloc))))
+  (some-> zloc z/up root?))
 
-(defn- top [zloc]
+(defn- root [zloc]
   (if (root? zloc) zloc (recur (z/up zloc))))
 
 (defn- clojure-whitespace? [zloc]
   (z/whitespace? zloc))
 
 (defn- surrounding-whitespace? [zloc]
-  (and (top? (z/up zloc))
+  (and (not (top? zloc))
```

Comment:
This was depending on the buggy definition of `top?`, which if it had been named according to it's implementation would have been called `non-root?`. The refactoring that arrived at this form went like this:
```clojure
(top? (z/up zloc))       ;; original, with buggy definition of top?
(non-root? (z/up zloc))  ;; rename top? to reflect actual behavior
(not (root? (z/up zloc)) ;; replace with negation of converse
(not (top? zloc))        ;; final, with new correct definition of top?
```